samtranslator/model/api/api_generator.py

```diff
@@ -93,6 +97,11 @@ def _construct_rest_api(self):
             raise InvalidResourceException(self.logical_id,
                                            "Specify either 'DefinitionUri' or 'DefinitionBody' property and not both")
 
+        if self.open_api_version:
+            if re.match(OpenApiVersionsSupportedRegex, self.open_api_version) is None:
+                raise InvalidResourceException(
+                    self.logical_id, "The OpenApiVersion value must be of the format 2.x.x or 3.x.x")
```

Comment:
"2.x.x" isn't valid in that regex, right? This is confusing because "2.x.x" is a commonly accepted value for semantic versions.

Reply:
2.x.x is a valid value for the flag. This is the regex - r"\A[2-3](\.\d)(\.\d)?$"

Reply:
'x' isn't a digit/number though. Maybe there's some Python thing I'm missing here.

Reply:
Got it, resolved. Thanks!